sccameraview/src/main/java/com/example/sccameraview/Camera2View.java

```diff
@@ -369,6 +379,70 @@ public int compare(Size lhs, Size rhs) {
         }
     }
 
+    @Override
+    public void takePicture() {
+        imageReader = ImageReader.newInstance(previewSize.getWidth(), previewSize.getHeight(), ImageFormat.JPEG, 1);
+        imageReader.setOnImageAvailableListener(mImageAvailableListener, backgroundHandler);
+
+        try {
+            final int sensorOrientation = cameraManager.getCameraCharacteristics(cameraIdString).get(
+                    CameraCharacteristics.SENSOR_ORIENTATION);
+            cameraDevice.createCaptureSession(Collections.singletonList(imageReader.getSurface()),
+                new CameraCaptureSession.StateCallback() {
+                    @Override
+                    public void onConfigured(@NonNull CameraCaptureSession session) {
+                        try {
+                        CaptureRequest.Builder captureRequestBuilder = cameraDevice.createCaptureRequest(
+                                CameraDevice.TEMPLATE_STILL_CAPTURE);
+                        captureRequestBuilder.addTarget(imageReader.getSurface());
+                        captureRequestBuilder.set(CaptureRequest.CONTROL_AF_MODE,
+                                CaptureRequest.CONTROL_AF_MODE_OFF);
```

Comment:
Autofocus and flash should be controlled by the client, buy using boolean setters.